### PR TITLE
Fix utils test names

### PR DESCRIPTION
**Pull request by esteffin**
Changes to make tests compatible with XUnit:
 - Removed special characters (`<`, `=`, `>`) from test names
 - Added flag to gtest to generate XUnit reports

### CODE REVIEW

File: utils/tests/coders.bats [hunk 1]
```diff
@@ -32,15 +32,15 @@ function teardown {
...
+@test "BGV: data equals decode(encode(data))" {
   run bash -c "$encode ${prefix_bgv}.dat ${prefix_bgv}.info BGV > ${prefix_bgv}.encoded"
   assert [ "$status" -eq 0 ]
   run bash -c "$decode ${prefix_bgv}.encoded ${prefix_bgv}.info BGV > ${prefix_bgv}.decoded"
   assert [ "$status" -eq 0 ]
   diff "${prefix_bgv}.dat" "${prefix_bgv}.decoded"
 }
 
-@test "BGV: data != decode(encode(data)) with padding" {
+@test "BGV: data not equals decode(encode(data)) with padding" {
```
Comment (jlhcrawford) [inline utils/tests/coders.bats:43]:
```suggestion
@test "BGV: data does not equal decode(encode(data)) with padding" {
```

File: utils/tests/coders.bats [hunk 2]
```diff
@@ -81,15 +81,15 @@ function teardown {
...
+@test "CKKS: data equals decode(encode(data))" {
   run bash -c "$encode ${prefix_ckks}.dat ${prefix_ckks}.info CKKS > ${prefix_ckks}.encoded"
   assert [ "$status" -eq 0 ]
   run bash -c "$decode ${prefix_ckks}.encoded ${prefix_ckks}.info CKKS > ${prefix_ckks}.decoded"
   assert [ "$status" -eq 0 ]
   diff "${prefix_ckks}.dat" "${prefix_ckks}.decoded"
 }
 
-@test "CKKS: data != decode(encode(data)) with padding" {
+@test "CKKS: data not equals decode(encode(data)) with padding" {
```
Comment (jlhcrawford) [inline utils/tests/coders.bats:92]:
```suggestion
@test "CKKS: data does not equal decode(encode(data)) with padding" {
```

File: utils/tests/create-context.bats [hunk 3]
```diff
@@ -115,19 +115,19 @@ function teardown {
   assert [ "$output" == "Frobenius matrices reqires switch-key matrices to be generated." ]
 }
 
-@test "default scheme fails if p < 2" {
+@test "default scheme fails if p less than 2" {
```
Comment (jlhcrawford) [inline utils/tests/create-context.bats:118]:
```suggestion
@test "default scheme fails if p is less than 2" {
```

File: utils/tests/create-context.bats [hunk 4]
```diff
@@ -115,19 +115,19 @@ function teardown {
...
   run "${create_context}" "${prefix_ckks}.params"
   assert [ "$status" -ne 0 ]
   assert [ "$output" == "BGV invalid plaintext modulus. In BGV it must be a prime number greater than 1." ]
 }
 
-@test "BGV: fails if p < 2" {
+@test "BGV: fails if p less than 2" {
```
Comment (jlhcrawford) [inline utils/tests/create-context.bats:124]:
```suggestion
@test "BGV: fails if p is less than 2" {
```

File: utils/tests/create-context.bats [hunk 5]
```diff
@@ -115,19 +115,19 @@ function teardown {
...
   run "${create_context}" "${prefix_ckks}.params" --scheme BGV
   assert [ "$status" -ne 0 ]
   assert [ "$output" == "BGV invalid plaintext modulus. In BGV it must be a prime number greater than 1." ]
 }
 
-@test "CKKS: fails if p != -1" {
+@test "CKKS: fails if p not equals -1" {
```
Comment (jlhcrawford) [inline utils/tests/create-context.bats:130]:
```suggestion
@test "CKKS: fails if p does not equal -1" {
```

File: utils/tests/crypto.bats [hunk 6]
```diff
@@ -55,70 +55,70 @@ function teardown {
...
+@test "decrypt fails with invalid nthreads less than 0" {
   run $encrypt "${pk_file_bgv}" "${prefix_bgv}.ptxt"
   assert [ "$status" -eq 0 ]
   run $decrypt "${sk_file_bgv}" "${prefix_bgv}.ctxt" -o result_bgv.decrypt -n -1
   assert [ "$status" -ne 0 ]
   assert [ "$output" == "Number of threads must a be positive integer." ]
 }
 
-@test "encrypt fails with invalid batch size = -1" {
+@test "encrypt fails with invalid batch size equals -1" {
```
Comment (jlhcrawford) [inline utils/tests/crypto.bats:100]:
```suggestion
@test "encrypt fails with invalid batch size equal to -1" {
```

File: utils/tests/crypto.bats [hunk 7]
```diff
@@ -55,70 +55,70 @@ function teardown {
...
   run $encrypt "${pk_file_bgv}" "${prefix_bgv}.ptxt" -b -1
   assert [ "$status" -ne 0 ]
   assert [ "$output" == "Batch size must be a positive integer." ]
 }
 
-@test "encrypt warns if batch size < nthreads" {
+@test "encrypt warns if batch size less than nthreads" {
```
Comment (jlhcrawford) [inline utils/tests/crypto.bats:106]:
```suggestion
@test "encrypt warns if batch size is less than nthreads" {
```

File: utils/tests/crypto.bats [hunk 8]
```diff
@@ -55,70 +55,70 @@ function teardown {
...
   run $encrypt "${pk_file_bgv}" "${prefix_bgv}.ptxt" -b 1 -n 2
   assert [ "$status" -eq 0 ]
   assert [ "${lines[0]}" == "WARNING: Not enough elements in the batch (1) to run with 2 threads." ]
   assert [ "${lines[1]}" == "To achieve better performance set batch size (-b) to be equal to or more than the number of threads (-n). Ideally the batch size should be a multiple of the number of threads." ]
 }
 
-@test "decrypt fails with invalid batch size = -1" {
+@test "decrypt fails with invalid batch size equals -1" {
```
Comment (jlhcrawford) [inline utils/tests/crypto.bats:113]:
```suggestion
@test "decrypt fails with invalid batch size equal to -1" {
```

File: utils/tests/crypto.bats [hunk 9]
```diff
@@ -55,70 +55,70 @@ function teardown {
...
   run $encrypt "${pk_file_bgv}" "${prefix_bgv}.ptxt"
   assert [ "$status" -eq 0 ]
   run $decrypt "${sk_file_bgv}" "${prefix_bgv}.ctxt" -o result_bgv.decrypt -b -1
   assert [ "$status" -ne 0 ]
   assert [ "$output" == "Batch size must be a positive integer." ]
 }
 
-@test "decrypt warns if batch size < nthreads" {
+@test "decrypt warns if batch size less than nthreads" {
```
Comment (jlhcrawford) [inline utils/tests/crypto.bats:121]:
```suggestion
@test "decrypt warns if batch size is less than nthreads" {
```

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
Looks fine, just some grammatical errors.

**Review by jlhcrawford — APPROVED** [review]

**Review by hamishun — APPROVED** [review]

**Pull request merged**